### PR TITLE
Add possibility to generate jobs specific repositories

### PR DESCRIPTION
**Pull request by radtriste**
Useful for testing only specific pipeline

### CODE REVIEW

**Comment by radtriste** [comment]
tested all ok and using it :)

File: docs/jenkins.md [hunk 1]
```diff
@@ -114,4 +109,34 @@ $ chmod u+x .jenkins/dsl/test.sh
 $ cd .jenkins/dsl && ./scripts/test.sh
 ```
 
-*NOTE: Your Jenkinsfiles should be in `.jenkins` folder but can be anywhere. The reference is anyway done in the job script, so you can put whatever you want if needed.*
\ No newline at end of file
+*NOTE: Your Jenkinsfiles should be in `.jenkins` folder but can be anywhere. The reference is anyway done in the job script, so you can put whatever you want if needed.*
```
Comment (Kaitou786) [inline docs/jenkins.md]: This Note seems confusing to me, I mean first it is written that `it should be in...` and then can be anywhere. can we rephrase this statement?

File: dsl/seed/Jenkinsfile.seed [hunk 2]
```diff
@@ -45,7 +45,21 @@ pipeline {
...
+                    if (getCustomBranchKey()) {
+                        currentBuild.displayName = getCustomBranchKey()
+
+                        assert getCustomRepositories()
+                        branchConfigs = [ "${getCustomBranchKey()}" : [:] ]
+                        if (getCustomMainBranch()) {
+                            seedConfig.git.main_branch = [ "default" : getCustomMainBranch() ]
+                        }
+                    } else {
+                        branchConfigs = seedConfig.git.branches.collectEntries { [it, [:]] }
```
Comment (Kaitou786) [inline dsl/seed/Jenkinsfile.seed:58]: Just out of curiosity what does this `it` object represent here?

**Review by Kaitou786 — COMMENTED** [review]
Also why do we have these lines commented here: https://github.com/kiegroup/kogito-pipelines/pull/115/files#diff-7fc4c71a02d66fa47c3dece6c2ae2e8745e18501011f2bbf08b4b45f3c5ba0e5R24-R26

File: dsl/seed/Jenkinsfile.seed (hunk 2 above)
Comment (radtriste) [reply dsl/seed/Jenkinsfile.seed:58]: `it` represent the iterative object.
you can do many ways in groovy. this way or:
```
seedConfig.git.branches.collectEntries { branch -> [branch, [:]] }
```
one other thing is about iteration on maps:
```
map.each { println "${it.key} = ${it.value}" }

map.each { entry -> println "${entry.key} = ${entry.value}" }

map.each { key, value -> println "${key} = ${value}" }
```
all are valid :D

File: docs/jenkins.md (hunk 1 above)
Comment (radtriste) [reply docs/jenkins.md]: updated

**Review by Kaitou786 — APPROVED** [review]

**Review by msmagnanijr — APPROVED** [review]
@radtriste nice work :)

**Pull request merged**